src/Microsoft.VisualStudio.Services.Agent/ThrottlingReportHandler.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Common.Internal;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public interface IThrottlingReporter
+    {
+        void ReportThrottling(string delay, string expiration);
+    }
+
+    public class ThrottlingReportHandler : DelegatingHandler
+    {
+        private IThrottlingReporter _throttlingReporter;
+
+        public ThrottlingReportHandler(IThrottlingReporter throttlingReporter)
+            : base()
+        {
+            ArgUtil.NotNull(throttlingReporter, nameof(throttlingReporter));
+            _throttlingReporter = throttlingReporter;
+        }
+
+        protected async override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            // Call the inner handler.
+            var response = await base.SendAsync(request, cancellationToken);
+
+            // Inspect whether response has throttling information
+            IEnumerable<string> vssRequestDelayed = null;
+            IEnumerable<string> vssRequestQuotaReset = null;
+            if (response.Headers.TryGetValues(HttpHeaders.VssRequestDelayed, out vssRequestDelayed) &&
+                response.Headers.TryGetValues(HttpHeaders.VssRequestQuotaReset, out vssRequestQuotaReset) &&
+                !string.IsNullOrEmpty(vssRequestDelayed.FirstOrDefault()) &&
+                !string.IsNullOrEmpty(vssRequestQuotaReset.FirstOrDefault()))
+            {
+                _throttlingReporter.ReportThrottling(vssRequestDelayed.First(), vssRequestQuotaReset.First());
```

Comment:
The delegating handler is what understands what these header values are (timespan, int of seconds, etc) so it should handle parsing them to the strong types and providing that to the throttling reporter rather than passing strings.